src/OrleansBondUtils/BondSerializer.cs

```diff
@@ -191,15 +197,35 @@ private void Register(Type type)
             var clonerType = typeof(Cloner<>);
             var realType = clonerType.MakeGenericType(type);
             var clonerInstance = Activator.CreateInstance(realType);
-            var cloneMethod = realType.GetMethod("Clone").MakeGenericMethod(type);
-            var copierDelegate = cloneMethod.CreateDelegate(
-                    typeof(Func<,>).MakeGenericType(new[] { type, type }),
-                    clonerInstance);
             var serializer = new BondTypeSerializer(type);
             var deserializer = new BondTypeDeserializer(type);
-            CopierDictionary.TryAdd(type.TypeHandle, copierDelegate);
+            var sourceParameter = Expression.Parameter(typeof(object));
+            var instanceParameter = Expression.Parameter(typeof(object));
+            var method = typeof(BondSerializer).GetMethod("DeepCopy", BindingFlags.NonPublic | BindingFlags.Static).MakeGenericMethod(type);
```

Comment:
yes. It's by design. I created the DeepCopy method at line 148 since the code was reusable. It makes the expression tree code smaller since it's only generating
```csharp
(source, instance) => DeepCopy((Type)source, instance)
```